packages/mdc-textfield/adapter.ts

```diff
@@ -0,0 +1,150 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import {EventType, SpecificEventListener} from '@material/base/index';
+import {NativeInputType} from './types';
+
+/**
+ * Defines the shape of the adapter expected by the foundation.
+ * Implement this adapter for your framework of choice to delegate updates to
+ * the component in your framework of choice. See architecture documentation
+ * for more details.
+ * https://github.com/material-components/material-components-web/blob/master/docs/code/architecture.md
+ */
+interface MDCTextFieldAdapter {
+  /**
+   * Adds a class to the root Element.
+   */
+  addClass(className: string): void;
+
+  /**
+   * Removes a class from the root Element.
+   */
+  removeClass(className: string): void;
+
+  /**
+   * @return true if the root element contains the given class name.
+   */
+  hasClass(className: string): boolean;
+
+  /**
+   * Registers an event handler on the root element for a given event.
+   */
+  registerTextFieldInteractionHandler<E extends EventType>(evtType: E, handler: SpecificEventListener<E>): void;
+
+  /**
+   * Deregisters an event handler on the root element for a given event.
+   */
+  deregisterTextFieldInteractionHandler<E extends EventType>(evtType: E, handler: SpecificEventListener<E>): void;
+
+  /**
+   * Registers an event listener on the native input element for a given event.
+   */
+  registerInputInteractionHandler<E extends EventType>(evtType: E, handler: SpecificEventListener<E>): void;
+
+  /**
+   * Deregisters an event listener on the native input element for a given event.
+   */
+  deregisterInputInteractionHandler<E extends EventType>(evtType: E, handler: SpecificEventListener<E>): void;
+
+  /**
+   * Registers a validation attribute change listener on the input element.
+   * Handler accepts list of attribute names.
+   */
+  registerValidationAttributeChangeHandler(handler: (attributeNames: string[]) => void): MutationObserver;
+
+  /**
+   * Disconnects a validation attribute observer on the input element.
+   */
+  deregisterValidationAttributeChangeHandler(observer: MutationObserver): void;
+
+  /**
+   * @return An object representing the native text input element, with a similar API shape.
+   *     We never alter the value within our code, however we do update the disabled
+   *     property, so if you choose to duck-type the return value for this method
+   *     in your implementation it's important to keep this in mind. Also note that
+   *     this method can return null, which the foundation will handle gracefully.
+   */
+  getNativeInput(): HTMLInputElement | NativeInputType | null;
```

Comment:
I think this should just extend from `NativeInputType`, and then have that extend from `HTMLInputElement`

Reply:
As we discussed offline, we're going to keep this as-is because we need to return a dummy object for unit tests.